### PR TITLE
feat(tier4_autoware_utils): add inset target point function by length

### PR DESCRIPTION
**Pull request by purewater0901**
## Description

Since the use can use current  insertPoint funcion only when we know the inserted point, it's not useful when we don't know the inserted point in advance.  To deal with this limitation, I added a new insertPoint function that only takes trajectory and insert point length. This function will be used insertStopPoint function in the future.

Note that insertStopPoint function has many implementations in autoware.universe now.

- [obstacle_cruise_planner](https://github.com/autowarefoundation/autoware.universe/blob/main/planning/obstacle_cruise_planner/src/utils.cpp#L156-L217)
- [behavior_path_planner](https://github.com/autowarefoundation/autoware.universe/blob/main/planning/behavior_path_planner/src/utilities.cpp#L1368-L1408)
- [motion_velocity_smoother](https://github.com/autowarefoundation/autoware.universe/blob/main/planning/motion_velocity_smoother/src/resample.cpp#L98-L119)
- [obstacle_stop_planner](https://github.com/autowarefoundation/autoware.universe/blob/main/planning/obstacle_stop_planner/src/node.cpp#L787)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by satoshi-ota** [comment]
https://github.com/autowarefoundation/autoware.universe/issues/596

File: common/tier4_autoware_utils/include/tier4_autoware_utils/trajectory/trajectory.hpp [hunk 1]
```diff
@@ -866,6 +866,47 @@ inline boost::optional<size_t> insertTargetPoint(
...
+{
+  validateNonEmpty(points);
+
+  if (insert_point_length < 0.0) {
+    return boost::none;
+  }
+
+  // Get Nearest segment index
+  boost::optional<size_t> segment_idx = boost::none;
+  for (size_t i = 1; i < points.size(); ++i) {
```
Comment (satoshi-ota) [inline common/tier4_autoware_utils/include/tier4_autoware_utils/trajectory/trajectory.hpp:888]: [imo] I think it is better to be able to the start index for the distance as an argument. What do you think ?

**Comment by satoshi-ota** [comment]
Could you show the example usage of this function or psuedo code in your `insertStopPoint()` ? :pray:

**Comment by yukkysaito** [comment]
Almost Ok :+1:

**Comment by purewater0901** [comment]
@satoshi-ota 
Example Usage
```
void insertStopPoint(const double lenght, T & points)
{
   idx = insertPoint(length, poitns)
   for(i=idx; i<points.size(); ++I)
   {
      points.vel = 0.0;
   }
}
```

**Pull request merged**